service/src/test/java/bio/terra/profile/pact/provider/BPMProviderTest.java

```diff
@@ -0,0 +1,122 @@
+import bio.terra.profile.service.iam.SamService;
+import bio.terra.profile.service.iam.model.SamResourceType;
+import bio.terra.profile.service.job.JobService;
+import bio.terra.profile.service.profile.exception.ProfileNotFoundException;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import bio.terra.profile.service.status.ProfileStatusService;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.jdbc.JdbcRepositoriesAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.cache.CacheManager;
+
+@Tag("provider-test")
+@Provider("bpm-provider")
+@PactBroker
+// for local testing, put any test pacts in the service/pacts folder.
+// then comment out the above line, and uncomment the following line
+// @PactFolder("pacts")
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = Main.class,
+    properties = {
+      "profile.profile-database.initialize-on-start=false",
+      "profile.profile-database.upgrade-on-start=false"
+    })
+@EnableAutoConfiguration(
+    exclude = {DataSourceAutoConfiguration.class, JdbcRepositoriesAutoConfiguration.class})
+public class BPMProviderTest {
+
+  @LocalServerPort int port;
+
+  @MockBean AzureCrlService azureCrlService;
+  @MockBean GcpCrlService gcpCrlService;
+  @MockBean ProfileDao profileDao;
+  @MockBean SamService samService;
+  @MockBean JobService jobService;
+
+  @MockBean ProfileStatusService profileStatusService;
+  @MockBean CacheManager cacheManager;
+  @MockBean UnauthenticatedApiController unauthenticatedApiController;
+
+  @MockBean AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @Mock AuthenticatedUserRequest userRequest;
+
+  @BeforeEach
+  void setUp(PactVerificationContext context) {
+    when(authenticatedUserRequestFactory.from(any())).thenReturn(userRequest);
+    context.setTarget(new HttpTestTarget("localhost", port, "/"));
+  }
+
+  @TestTemplate
+  @ExtendWith(PactVerificationSpringProvider.class)
+  void verifyPact(PactVerificationContext context) {
+    context.verifyInteraction();
+  }
+
+  @State("a GCP billing profile")
+  Map<String, Object> gcpBillingProfileState() throws InterruptedException {
+    var profile = ProviderStateData.gcpBillingProfile;
+    setUpProfileDaoGets(List.of(profile));
+    when(samService.hasActions(any(), eq(SamResourceType.PROFILE), eq(profile.id())))
+        .thenReturn(true);
+    return Map.of(
```

Comment:
To see if I'm understanding… this is saying that when a consumer asks for a GCP Billing profile, the only things that it uses in the response are the `gcpProfileId` and the `billingAccountId`?

Reply:
No; It just means that these are specific values that are available in the consumer pact.

For instance, if you are testing retrieving the profile by id (which in fact is what the WSM tests are doing), you want to ensure that the returned object has the same id as what you requested.
But you need to know what id to query. So if you specify `"${gcpProfileId}"` in the consumer test, this is what you'll get. Then you know both what id to use in the query, and what use for comparisons.

Reply:
But isn't something supposed to verify that the provider returns the parameters that the code depends on?